suite/tests/linux/sigmangle_asynch.c

```diff
@@ -0,0 +1,172 @@
+/* **********************************************************
+ * Copyright (c) 2015-2018 Google, Inc.  All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Google, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL GOOGLE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+#ifndef ASM_CODE_ONLY /* C code */
+#    include "tools.h"
+#    include "sigmangle_asynch-shared.h"
+#    include <setjmp.h>
+#    include <signal.h>
+#    include <pthread.h>
+
+volatile bool test_ready = false;
+volatile bool test_done = false;
+volatile int loop_inc = 1;
+
+/* asm routines */
+void
+test_asm();
+
+static SIGJMP_BUF mark;
+
+static void
+handle_signal(int signal, siginfo_t *siginfo, ucontext_t *ucxt)
+{
+#    ifdef X86_64
+    if (signal == SIGILL) {
+        sigcontext_t *sc = SIGCXT_FROM_UCXT(ucxt);
+        if (sc->LOOP_COUNT_REG_SIG != LOOP_COUNT)
+            print("ERROR: incorrect result!\n");
+        SIGLONGJMP(mark, 1);
+    } else if (signal != SIGUSR2) {
+        print("Unexpected signal!\n");
+    }
+#    endif
+}
+
+static void *
+thread_routine(void *arg)
+{
+#    ifdef X86_64
+    pthread_t main_thread = *(pthread_t *)arg;
+    while (!test_ready)
+        ;
```

Comment:
style: Add  a comment `/* Empty. */` or sthg so we know this is deliberate.

Reply:
Done